pkg/handlers/config.go

```diff
@@ -228,88 +230,103 @@ func (h *Handler) LiveAppConfig(w http.ResponseWriter, r *http.Request) {
+			return
+		}
+		defer os.RemoveAll(archiveDir)
 
-	// get values from request
-	configValues := configValuesFromConfigGroups(liveAppConfigRequest.ConfigGroups)
+		err = store.GetStore().GetAppVersionArchive(foundApp.ID, liveAppConfigRequest.Sequence, archiveDir)
+		if err != nil {
+			logger.Error(err)
+			liveAppConfigResponse.Error = "failed to get app version archive"
+			JSON(w, http.StatusInternalServerError, liveAppConfigResponse)
+			return
+		}
 
-	registryInfo, err := store.GetStore().GetRegistryDetailsForApp(foundApp.ID)
-	if err != nil {
-		logger.Error(err)
-		liveAppConfigResponse.Error = "failed to get app registry info"
-		JSON(w, http.StatusInternalServerError, liveAppConfigResponse)
-		return
-	}
+		kotsKinds, err = kotsutil.LoadKotsKindsFromPath(archiveDir)
+		if err != nil {
+			logger.Error(err)
+			liveAppConfigResponse.Error = "failed to load kots kinds from path"
+			JSON(w, http.StatusInternalServerError, liveAppConfigResponse)
+			return
+		}
 
-	localRegistry := template.LocalRegistry{
-		Host:      registryInfo.Hostname,
-		Namespace: registryInfo.Namespace,
-		Username:  registryInfo.Username,
-		Password:  registryInfo.Password,
-		ReadOnly:  registryInfo.IsReadOnly,
+		registryInfo, err := store.GetStore().GetRegistryDetailsForApp(foundApp.ID)
+		if err != nil {
+			logger.Error(err)
+			liveAppConfigResponse.Error = "failed to get app registry info"
+			JSON(w, http.StatusInternalServerError, liveAppConfigResponse)
+			return
+		}
+
+		localRegistry = template.LocalRegistry{
+			Host:      registryInfo.Hostname,
+			Namespace: registryInfo.Namespace,
+			Username:  registryInfo.Username,
+			Password:  registryInfo.Password,
+			ReadOnly:  registryInfo.IsReadOnly,
+		}
 	}
 
-	versionInfo := template.VersionInfoFromInstallation(liveAppConfigRequest.Sequence+1, foundApp.IsAirgap, kotsKinds.Installation.Spec) // sequence +1 because the sequence will be incremented on save (and we want the preview to be accurate)
-	appInfo := template.ApplicationInfo{Slug: foundApp.Slug}
-	renderedConfig, err := kotsconfig.TemplateConfigObjects(kotsKinds.Config, configValues, appLicense, &kotsKinds.KotsApplication, localRegistry, &versionInfo, &appInfo, kotsKinds.IdentityConfig, util.PodNamespace, false)
+	versionInfo := template.VersionInfoFromInstallation(liveAppConfigRequest.Sequence+1, app.GetIsAirgap(), kotsKinds.Installation.Spec) // sequence +1 because the sequence will be incremented on save (and we want the preview to be accurate)
+	appInfo := template.ApplicationInfo{Slug: app.GetSlug()}
+	renderedConfig, err := kotsconfig.TemplateConfigObjects(kotsKinds.Config, configValues, appLicense, &kotsKinds.KotsApplication, localRegistry, &versionInfo, &appInfo, kotsKinds.IdentityConfig, app.GetNamespace(), false)
 	if err != nil {
 		logger.Error(err)
 		liveAppConfigResponse.Error = "failed to render templates"
 		JSON(w, http.StatusInternalServerError, liveAppConfigResponse)
 		return
 	}
 
-	if renderedConfig != nil {
```

Comment:
lets keep a nil check, `kotsconfig.TemplateConfigObjects` can return a nil err and nil Config object.